andela/resly
# #102342692 cancel reservation

#### What does this PR do?

This PR enables a resly user to cancel a reservation.
#### Description of Task to be completed?

The user can cancel a reservation which he/her has already paid for.
This makes a new refund tied to that user and to that particular reservation.
The refund is a percentage of the total cost of that reservation.
#### How should this be manually tested?

After login and creating a reservation,

> Landing page - Your profile.
> Make a reservation and visit the restaurant.
> Current Reservations.
> Hit the `x` button to cancel a reservation.
> Confirm that the reservation is to be cancelled.
> Goto Cancelled reservations page and view the cancelled reservations.
#### Any background context you want to provide?

A user that is not able to make it for his reservation might wish to cancel it. This is now possible but the user will only be refunded a percentage of the cost of the reservation.
A user cannot cancel a reservation that is not tied to himself/herself.
A user cannot cancel a reservation that is due in the next 15 minutes.
A user cannot cancel a reservation that is passed.
#### What are the relevant pivotal tracker stories?

The Pivotal Tracker for this story can be viewed [here](https://www.pivotaltracker.com/projects/1415814/stories/102342692).
#### Screenshots (if appropriate)

![screen shot 2016-04-11 at 11 44 23 pm](https://cloud.githubusercontent.com/assets/17028608/14462841/bf96b7da-00c0-11e6-866b-de7ff9a31ab4.png)
![screen shot 2016-04-11 at 11 44 38 pm](https://cloud.githubusercontent.com/assets/17028608/14462846/bf989c1c-00c0-11e6-96d3-f840924502b3.png)
![screen shot 2016-04-11 at 11 45 07 pm](https://cloud.githubusercontent.com/assets/17028608/14462842/bf97083e-00c0-11e6-8ef6-7ea204c44af1.png)
![screen shot 2016-04-11 at 11 45 34 pm](https://cloud.githubusercontent.com/assets/17028608/14462845/bf97b9e6-00c0-11e6-80c0-fe820bc2a8fd.png)
![screen shot 2016-04-11 at 11 46 17 pm](https://cloud.githubusercontent.com/assets/17028608/14462843/bf97664e-00c0-11e6-9bdb-ce4f9a196169.png)
![screen shot 2016-04-11 at 11 46 45 pm](https://cloud.githubusercontent.com/assets/17028608/14462844/bf97afbe-00c0-11e6-8c6a-61619fb25826.png)
![screen shot 2016-04-12 at 3 13 21 pm](https://cloud.githubusercontent.com/assets/17028608/14462989/4ad1af6c-00c1-11e6-93da-b6bd05b2a529.png)
#### Questions:

There are no questions at this time.
